File: Assets/CSJ/Scripts/Api/Mediation/MediationPrivacyConfig.cs
namespace ByteDance.Union.Mediation
{
    using System.Collections.Generic;

    public class MediationPrivacyConfig
    {
        /// <summary>
        ///  仅android支持
        /// </summary>
        public List<string> CustomAppList { get; set; }
        /// <summary>
        ///  仅android支持
        /// </summary>
        public List<string> CustomDevImeis { get; set; }

        /// <summary>
        ///  仅android支持，默认值true.
        /// </summary>
        public bool CanUseOaid { get; set; } = true;

        /// <summary>
        /// 是否限制个性化广告，需adn支持，默认值false.
        /// </summary>
        public bool LimitPersonalAds { get; set; } = false;

        /// <summary>
        /// 是否程序化广告，需adn支持，仅iOS支持，默认值true.
        /// </summary>
        public bool ProgrammaticRecommend { get; set; } = true;

    }
}

File: Assets/CSJ/Plugins/Android/CSJ.androidlib/src/com/bytedance/android/MediationPrivacyConfig.java
package com.bytedance.android;

import com.bytedance.sdk.openadsdk.mediation.init.IMediationPrivacyConfig;

import java.util.List;

public class MediationPrivacyConfig implements IMediationPrivacyConfig {

    private List<String> customAppList;
    private List<String> customDevImeis;
    private boolean canUseOaid = true;
    private boolean limitPersonalAds = false;
    private boolean programmaticRecommend = true;

    public MediationPrivacyConfig() {
    }

    public void setCustomAppList(List<String> customAppList) {
        this.customAppList = customAppList;
    }

    public void setCustomDevImeis(List<String> customDevImeis) {
        this.customDevImeis = customDevImeis;
    }

    public void setCanUseOaid(boolean canUseOaid) {
        this.canUseOaid = canUseOaid;
    }

    public void setLimitPersonalAds(boolean limitPersonalAds) {
        this.limitPersonalAds = limitPersonalAds;
    }

    public void setProgrammaticRecommend(boolean programmaticRecommend) {
        this.programmaticRecommend = programmaticRecommend;
    }

    public List<String> getCustomAppList() {
        return customAppList;
    }

    public List<String> getCustomDevImeis() {
        return customDevImeis;
    }

    public boolean isCanUseOaid() {
        return canUseOaid;
    }

    public boolean isLimitPersonalAds() {
        return limitPersonalAds;
    }

    public boolean isProgrammaticRecommend() {
        return programmaticRecommend;
    }
}
